play/src/front/Components/MapEditor/EntityEditor/EntityUpload/EntityUpload.svelte

```diff
@@ -0,0 +1,139 @@
+    import { Direction, EntityPrefab } from "@workadventure/map-editor";
+    import LL from "../../../../../i18n/i18n-svelte";
+    import { mapEditorEntityUploadEventStore } from "../../../../Stores/MapEditorStore";
+    import CustomEntityEditionForm from "../CustomEntityEditionForm/CustomEntityEditionForm.svelte";
+
+    let files: FileList | undefined = undefined;
+    let dropZoneRef: HTMLDivElement;
+    let customEntityToUpload: EntityPrefab | undefined = undefined;
+    const allowedImageFormats = ["image/png", "image/jpeg", "image/webp"];
+
+    $: {
+        if (files) {
+            customEntityToUpload = {
+                collectionName: "custom entities",
+                name: files.item(0)!.name,
+                imagePath: URL.createObjectURL(files.item(0)!),
+                id: uuidv4(),
+                direction: Direction.Down,
+                tags: [],
+                color: "",
+                type: "Custom",
+            };
+        }
+    }
+
+    const mapEditorEntityUploadEventStoreUnsubscriber = mapEditorEntityUploadEventStore.subscribe(
+        (uploadEntityMessage) => {
+            completeAndResetUpload(uploadEntityMessage);
+        }
+    );
+
+    function completeAndResetUpload(uploadEntityMessage: UploadEntityMessage | undefined) {
+        if (uploadEntityMessage === undefined && files !== undefined) {
+            initFileUpload();
+        }
+    }
+
+    async function processFileToUpload(customEditedEntity: EntityPrefab) {
+        const fileToUpload = files?.item(0);
+        if (fileToUpload) {
+            const fileBuffer = await fileToUpload.arrayBuffer();
+            const fileAsUint8Array = new Uint8Array(fileBuffer);
+            const generatedId = uuidv4();
+            mapEditorEntityUploadEventStore.set({
+                id: generatedId,
+                file: fileAsUint8Array,
+                direction: CustomEntityDirection.Down,
+                name: customEditedEntity.name,
+                tags: customEditedEntity.tags,
+                imagePath: `${generatedId}-${fileToUpload.name}`,
+                collisionGrid: customEditedEntity.collisionGrid,
+                depthOffset: customEditedEntity.depthOffset,
+                color: "",
+            });
+        }
+    }
+
+    function initFileUpload() {
+        files = undefined;
+        customEntityToUpload = undefined;
+        mapEditorEntityUploadEventStore.set(undefined);
+    }
+
+    function dropHandler(event: DragEvent) {
+        const { files: filesFromDropEvent } = event.dataTransfer ?? {};
+        if (filesFromDropEvent) {
+            if (filesFromDropEvent.length > 1) {
+                console.error("Only one file is permitted");
+            } else {
+                if (allowedImageFormats.includes(filesFromDropEvent.item(0)?.type ?? "")) {
+                    files = filesFromDropEvent;
+                } else {
+                    console.error("File format not supported");
```

Comment:
Let's put a small error in the component here to give some visual feedback to the user.